backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPoint.java

```diff
@@ -208,6 +203,58 @@ Not part of FDP specification but good practice (https://specs.fairdatapoint.org
     return stringWriter.toString();
   }
 
+  private static void iterateOverSchemasAddPropVal(
+      Map<String, List<Map<String, Object>>> allCatalogFromJSON,
+      String apiFdpCatalog,
+      ModelBuilder builder,
+      IRI apiFdpCatalogEnc,
+      IRI apiFdpEnc)
+      throws Exception {
+    for (String schemaName : allCatalogFromJSON.keySet()) {
+      propValsPerSchema(
+          allCatalogFromJSON, apiFdpCatalog, builder, apiFdpCatalogEnc, apiFdpEnc, schemaName);
+    }
+  }
+
+  private static void propValsPerSchema(
+      Map<String, List<Map<String, Object>>> allCatalogFromJSON,
+      String apiFdpCatalog,
+      ModelBuilder builder,
+      IRI apiFdpCatalogEnc,
+      IRI apiFdpEnc,
+      String schemaName)
+      throws Exception {
+    for (Map<String, Object> map : allCatalogFromJSON.get(schemaName)) {
+      IRI catalogIriEnc = encodedIRI(apiFdpCatalog + "/" + schemaName + "/" + map.get("id"));
+      builder.add(apiFdpCatalogEnc, iri("http://www.w3.org/ns/ldp#contains"), catalogIriEnc);
+      builder.add(apiFdpEnc, iri("https://w3id.org/fdp/fdp-o#metadataCatalog"), catalogIriEnc);
+      nullCheckOnPropVal(builder, map, catalogIriEnc);
+    }
+  }
+
+  private static void nullCheckOnPropVal(
+      ModelBuilder builder, Map<String, Object> map, IRI catalogIriEnc) throws Exception {
+    if (map.get("propertyValue") != null) {
+      splitPropValAndAddToBuilder(builder, map, catalogIriEnc);
+    }
+  }
+
+  private static void splitPropValAndAddToBuilder(
+      ModelBuilder builder, Map<String, Object> map, IRI catalogIriEnc) throws Exception {
+    for (String propertyValue : (List<String>) map.get("propertyValue")) {
+      String[] propertyValueSplit = propertyValue.split(" ", -1);
+      checkPropValSplitLength(propertyValueSplit);
+      builder.add(catalogIriEnc, iri(propertyValueSplit[0]), iri(propertyValueSplit[1]));
+    }
+  }
+
+  private static void checkPropValSplitLength(String[] propertyValueSplit) throws Exception {
```

Comment:
IllegalArgumentException is an unchecked exception and doesn't require a throws declaration.

Reply:
well spotted